libraries/AP_NavEKF3/AP_NavEKF3_VehicleStatus.cpp

```diff
@@ -83,16 +83,6 @@ void NavEKF3_core::calcGpsGoodToAlign(void)
         gpsVertVelFilt = 0.1f * gpsDataNew.vel.z + 0.9f * gpsVertVelFilt;
         gpsVertVelFilt = constrain_float(gpsVertVelFilt,-10.0f,10.0f);
         gpsVertVelFail = (fabsf(gpsVertVelFilt) > 0.3f*checkScaler) && (frontend->_gpsCheck & MASK_GPS_VERT_SPD);
-    } else if (frontend->sources.useVelZSource(AP_NavEKF_Source::SourceZ::GPS) && !gps.have_vertical_velocity(preferred_gps)) {
```

Comment:
I don't get this bit.  It is still possible to configure the sources so you're using GPS as a vertical velocity source, and it is still possible for the GPS to not be able to supply it.  But now in that case we're going to set `gpsVertVelFail` to false - presumably indicating we have a valid source...

I can understand not flipping the parameters based on the GPS status, but don't we still need to say we don't have vertical velocity?

Reply:
gpsVertVelFail is a local stack var used to determine if we should consider GPS vertical velocity in determining if the GPS is ready for use. If the GPS doesn't have vert vel then it isn't "failed", ie. we don't wait for it to give us vertical velocity before we allow the EKF to start